src/Cache/Engine/RedisEngine.php

```diff
@@ -123,47 +121,84 @@ protected function _connect()
     /**
      * Write data for key into cache.
      *
+     * @deprecated Since 3.6 use set() instead
      * @param string $key Identifier for the data
      * @param mixed $value Data to be cached
      * @return bool True if the data was successfully cached, false on failure
      */
     public function write($key, $value)
     {
         $key = $this->_key($key);
-
         if (!is_int($value)) {
             $value = serialize($value);
         }
-
         $duration = $this->_config['duration'];
         if ($duration === 0) {
             return $this->_Redis->set($key, $value);
         }
-
         return $this->_Redis->setex($key, $duration, $value);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function set($key, $data, $ttl = null)
+    {
+        $key = $this->_key($key);
+
+        if (!is_int($data)) {
+            $data = serialize($data);
+        }
+
+        $duration = $this->_config['duration'];
+        if ($duration === 0) {
+            return $this->_Redis->set($key, $data);
+        }
+
+        return $this->_Redis->setex($key, $duration, $data);
+    }
+
     /**
      * Read a key from the cache
      *
+     * @deprecated Since 3.6 use set() instead
      * @param string $key Identifier for the data
      * @return mixed The cached data, or false if the data doesn't exist, has expired, or if there was an error fetching it
      */
     public function read($key)
     {
         $key = $this->_key($key);
-
         $value = $this->_Redis->get($key);
         if (preg_match('/^[-]?\d+$/', $value)) {
             return (int)$value;
         }
         if ($value !== false && is_string($value)) {
             return unserialize($value);
         }
-
         return $value;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function get($key, $default = null)
+    {
+        $key = $this->_key($key);
+
+        $value = $this->_Redis->get($key);
+        if (preg_match('/^[-]?\d+$/', $value)) {
+            return (int)$value;
+        }
+
+        if ($value !== false && is_string($value)) {
+            $result = unserialize($value);
+
+            return $result === false ? null : $result;
+        }
+
+        return $value === false ? null : $value;
```

Comment:
We aren't using the default value in this method, we should return that on failures.